src/rideables/Stack.hpp

```diff
@@ -0,0 +1,71 @@
+// #ifndef STACK_HPP
+// #define STACK_HPP
+
+#include <stdio.h>
+#include<iostream>
+#include <atomic>
+
+using namespace std;
+
+class Stack
+{
+    private:
+
+    typedef struct StackNode
+    {
+
+        int data;
+        StackNode *next;
+
+    } * StackNodePtr;
+
+    atomic<StackNodePtr> top;
+
+    public:
+    Stack()
+    {
+        top = NULL;
+    }
+    void push(int data){
+
+        StackNodePtr new_node = new StackNode;
+        new_node->data = data;
+        new_node->next = NULL;
+
+        StackNodePtr old_node;
+
+        do {
+            old_node = top;
+            new_node->next = old_node;
+        } while (!top.compare_exchange_strong(old_node, new_node));
+
+
+    }
+    int pop(){
+        StackNodePtr new_node;
+        StackNodePtr old_node;
+        do {
+            old_node = top;
+            if (old_node == NULL){
+                return NULL;
+            }
+            new_node = old_node->next;
+        } while (!top.compare_exchange_strong(old_node, new_node));
+        return old_node->data;
```

Comment:
Like C, C++ is manually memory managed, so you need to call `delete` to avoid memory leaks when _it's not in use anymore_; this is different from what's in Java, unfortunately. So technically, `old_node` here is a memory leak. 

However, directly calling `delete` _here_ may reclaim it while someone else may be still using it, so we need SMR..